gr-audio/lib/windows/windows_source.cc

```diff
@@ -181,11 +183,14 @@ int windows_source::work(int noutput_items,
                 dw_items += buffer_length / 2;
             }
             buffer_queue.pop();
+            lock.unlock();
```

Comment:
For safety, should this section be a scoped_lock?  What happens if there is an exception in those methods?

Reply:
What safety issue did you have in mind?

My understanding is that if an exception occurs, then the `unique_lock` destructor well be called, thereby unlocking it.